### DISCUSSION TITLE
Has anyone gotten Smart Lottery to work with the latest version of OpenZeppelin and Chainlink?

### DISCUSSION THREAD

**Discussion by freddie71010** [smartcontractkit/full-blockchain-solidity-course-py#955] [Q&A] [answered]
I've been trying to upgrade the tutorial to use the latest versions of OpenZeppelin and Chainlink contracts. The biggest issue I'm finding is related to the Mock files we create while doing our development testing of the `Lottery.sol` contract. Those don't work with the latest versions.  This is what I have right now:
```
pragma solidity ^0.7.0;

import "@chainlink/contracts/src/v0.7/interfaces/AggregatorV3Interface.sol";
import "@chainlink/contracts/src/v0.7/VRFConsumerBase.sol";
import "@openzeppelin/contracts/access/Ownable.sol";
```
My next step is to create a new version of the `LinkToken.sol`.  It's a [very nested contract running 0.4.11](https://github.com/smartcontractkit/chainlink-mix/blob/master/contracts/test/LinkToken.sol) so I'm trying to flatten it into one file so I can run that using `solidity 0.7.0`.  

I wanted to get others' opinions on all of this. Has anyone successfully updated the mock files (specifically the LinkToken.sol contract) to run using a higher version of solidity that what is found in the tutorial? You can take a look at my progress in [my github here](https://github.com/freddie71010/solidity-scripts/tree/smartcontract_lottery/smartcontract-lottery).

**Comment by cromewar** [comment]
Hello @freddie71010 are you using the latest version on chainlink repo o the one on chainlink-brownie-contracts?
I've made something like that on a project using a updated Mock for VRFCoordinator and using the Link token on version 0.4.11 (this token has no need to be upgraded), you can find that repo [here](https://github.com/cromewar/World-of-Hackathon-Wingbirds).

**Reply by freddie71010** [reply]
<img width="719" alt="image" src="https://user-images.githubusercontent.com/19274376/152875664-b4276063-c0f8-413e-ab8e-7bf3fcadacf9.png">
I copied your code from ur repo for LinkToken.sol and updated my brownie-config.yaml. I set that repo as `chainlink-brownie`.

brownie-config.yaml
```solidity
dependencies:
  - freddie71010/LinkToken@1.0.0
  - smartcontractkit/chainlink@1.1.0
  - smartcontractkit/chainlink-brownie-contracts@0.3.1
  - OpenZeppelin/openzeppelin-contracts@3.4.2
dotenv: .env
compiler:
  solc:
    version: 0.7.0
    remappings:
      - "@linktoken=freddie71010/LinkToken@1.0.0"
      - "@chainlink=smartcontractkit/chainlink@1.1.0"
      - "@chainlink-brownie=smartcontractkit/chainlink-brownie-contracts@0.3.1"
      - "@openzeppelin=OpenZeppelin/openzeppelin-contracts@3.4.2"
```

I'm getting pretty much the same error I've been getting - issues with the source file compiler version. Any ideas?

**Comment by cromewar** [comment]
As we already talked privately and the error is solved, I'm closing this thread now, however @freddie71010  could you please share the solution for the community?

**Comment by freddie71010** [comment, marked as answer]
@cromewar and I figured out the issue. It seems there's an issue with Brownie. In `Lottery.sol`, `brownie compile` was reading my commented out `pragma solidity ^0.6.6;` declaration instead of my declared `pragma solidity ^0.7.0;` statement. Once I removed the commented out code, I was able to compile my project even with the LinkToken.sol script that's running `pragma solidity ^0.4.11;`.

Broken:
```solidity
// SPDX-License-Identifier: MIT

// pragma solidity ^0.6.6;
// import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";

pragma solidity ^0.7.0;

import "@chainlink/contracts/src/v0.7/interfaces/AggregatorV3Interface.sol";
import "@chainlink/contracts/src/v0.7/VRFConsumerBase.sol";
import "@openzeppelin/contracts/access/Ownable.sol";

contract Lottery is VRFConsumerBase, Ownable {
....
}
```
<img width="801" alt="pragma_error" src="https://user-images.githubusercontent.com/19274376/153034255-4bd6b1ee-72f9-4267-95b5-bb06a79b5bb4.png">


Rerunning `brownie compile` after removing the commented out pragma statement - it works!:
```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.7.0;

import "@chainlink/contracts/src/v0.7/interfaces/AggregatorV3Interface.sol";
import "@chainlink/contracts/src/v0.7/VRFConsumerBase.sol";
import "@openzeppelin/contracts/access/Ownable.sol";

contract Lottery is VRFConsumerBase, Ownable {
....
}
```
<img width="794" alt="pragma_success2" src="https://user-images.githubusercontent.com/19274376/153034265-69b810af-d1d7-4ddb-8794-7fb2b56e754a.png">
